code/modules/holiday/halloween/bartholomew.dm

```diff
@@ -0,0 +1,148 @@
+/obj/effect/landmark/barthpot
+    name = "barthpot"
+
+/obj/item/barthpot
+    name = "Bartholomew"
+    icon = 'icons/obj/halloween_items.dmi'
+    icon_state = "barthpot"
+    anchored = TRUE
+    var/items_list = list()
+    speech_span = "spooky"
+    var/active = TRUE
+
+/obj/item/barthpot/Destroy()
+    var/obj/item/barthpot/n = new src(loc)
+    n.items_list = items_list
+    ..()
+
+
+/obj/item/barthpot/attackby(obj/item/I, mob/user, params)
+    if(!active)
+        say("Meow!")
+        return
+    for(var/I2 in items_list)
+        if(istype(I, I2))
+            qdel(I)
+            new /obj/item/reagent_containers/food/snacks/special_candy(loc)
+            to_chat(user, "<span class='notice'>You add the [I.name] to the pot and watch as it melts into the mixture, a candy crystalising in it's wake.</span>")
+            say("Hooray! Thank you!")
+            items_list -= I2
+            return
+    say("It doesn't seem like that's magical enough!")
+
+/obj/item/barthpot/attack_hand(mob/user)
+    if(!active)
+        say("Meow!")
+        return
+    say("Hello there, I'm Bartholomew, Jacqueline's Familiar.")
+    sleep(20)
+    say("I'm currently seeking items to put into my pot, if we get the right items, it should crystalise into a magic candy!")
+    if(!iscarbon(user))
+        say("Though... I'm not sure you can help me.")
+
+    var/message = "From what I can tell, "
+    if(LAZYLEN(items_list) < 5)
+        generate_items()
+    for(var/I2 in items_list)
+        if(!I2)
+            items_list -= I2
+            continue
+        var/obj/item/I3 = new I2
+        message += "a [I3.name], "
+    message += "currently seem to have the most magic potential."
+    sleep(15)
+    say("[message]")
+
+/obj/item/barthpot/proc/generate_items()
+    var/length = LAZYLEN(items_list)
+    var/rand_items = list(/obj/item/bodybag = 1,
+	/obj/item/clothing/glasses/meson = 2,
+	/obj/item/clothing/glasses/sunglasses = 1,
+	/obj/item/clothing/gloves/color/fyellow = 1,
+	/obj/item/clothing/head/hardhat = 1,
+	/obj/item/clothing/head/hardhat/red = 1,
+	/obj/item/clothing/head/that = 1,
+	/obj/item/clothing/head/ushanka = 1,
+	/obj/item/clothing/head/welding = 1,
+	/obj/item/clothing/mask/gas = 15,
+	/obj/item/clothing/suit/hazardvest = 1,
+	/obj/item/clothing/under/rank/vice = 1,
+	/obj/item/clothing/suit/hooded/flashsuit = 2,
+	/obj/item/clothing/accessory/medal/greytide	=	1,
+	/obj/item/assembly/prox_sensor = 4,
+	/obj/item/assembly/timer = 3,
+	/obj/item/flashlight = 4,
+	/obj/item/flashlight/pen = 1,
+	/obj/effect/spawner/lootdrop/glowstick = 4,
+	/obj/effect/spawner/lootdrop/mre = 3,
```

Comment:
Spawners don't grab the type of what you want!

Reply:
oops